src/Containers.jsx

```diff
@@ -590,19 +591,64 @@ class Containers extends React.Component {
         this.setState({ showPruneUnusedContainersModal: true });
     };
 
-    filterContainersByText = (lcf, id) => {
-        const container = this.props.containers[id];
-        const systemd_unit_match = container.Config?.Labels?.PODMAN_SYSTEMD_UNIT?.toLowerCase().indexOf(lcf) >= 0;
-        const name_match = container.Name.toLowerCase().indexOf(lcf) >= 0;
-        const image_match = container.ImageName.toLowerCase().indexOf(lcf) >= 0;
+    filterContainers = (containers) => {
+        let filtered = [];
+        filtered = Object.keys(containers).filter(id => !(this.props.filter == "running") || ["running", "restarting"].includes(this.props.containers[id]?.State?.Status));
+
+        const filter_by_text = (lcf, id) => {
+            const container = containers[id];
+            const systemd_unit_match = container.Config?.Labels?.PODMAN_SYSTEMD_UNIT?.toLowerCase().indexOf(lcf) >= 0;
+            const name_match = container.Name.toLowerCase().indexOf(lcf) >= 0;
+            const image_match = container.ImageName.toLowerCase().indexOf(lcf) >= 0;
+
+            if (container.Pod) {
+                const podKey = utils.makeKey(container.uid, container.Pod);
+                const pod = this.props.pods[podKey] || this.props.quadletPods[podKey];
+                const pod_match = pod.Name.toLowerCase().indexOf(lcf) >= 0 || pod.Labels?.PODMAN_SYSTEMD_UNIT?.toLowerCase().indexOf(lcf) >= 0;
+                return name_match || systemd_unit_match || image_match || pod_match;
+            } else {
+                return name_match || systemd_unit_match || image_match;
+            }
+        };
 
-        if (container.Pod) {
-            const pod = this.props.pods[utils.makeKey(container.uid, container.Pod)];
-            const pod_match = pod.Name.toLowerCase().indexOf(lcf) >= 0 || pod.Labels?.PODMAN_SYSTEMD_UNIT?.toLowerCase().indexOf(lcf) >= 0;
-            return name_match || systemd_unit_match || image_match || pod_match;
-        } else {
-            return name_match || systemd_unit_match || image_match;
+        if (this.props.ownerFilter !== "all") {
+            filtered = filtered.filter(id => {
+                if (this.props.ownerFilter === "user")
+                    return containers[id].uid === null;
+                return containers[id].uid === this.props.ownerFilter;
+            });
+        }
+
+        if (this.props.textFilter.length > 0) {
+            const lcf = this.props.textFilter.toLowerCase();
+            filtered = filtered.filter(id => filter_by_text(lcf, id));
         }
+
+        // Remove infra and service containers
+        filtered = filtered.filter(id => !containers[id].IsInfra && !containers[id].IsService);
+
+        const getHealth = id => {
+            const state = containers[id]?.State;
+            return state?.Health?.Status || state?.Healthcheck?.Status;
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2266-91a6840b-20250918-095040-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)